tests/src/core/testqgsexpression.cpp

```diff
@@ -2443,6 +2443,43 @@ class TestQgsExpression: public QObject
       QCOMPARE( badArray.evalErrorString(), QString( "Cannot convert 'not an array' to array" ) );
     }
 
+    void compare_arrays()
+    {
+      QCOMPARE( QgsExpression( "array() = array()" ).evaluate(), QVariant( true ) );
+      QCOMPARE( QgsExpression( "array(NULL) = array(NULL)" ).evaluate(), QVariant( true ) );
```

Comment:
This is odd - I'd have expected the result to be null here (to match null=null). But you're right, Postgres reports this as true, so leave it this way.